src/sentry/models/groupsubscription.py

```diff
@@ -110,10 +115,13 @@ def get_participants(self, group: "Group") -> Mapping[ExternalProviders, Mapping
         Identify all users who are participating with a given issue.
         :param group: Group object
         """
-        from sentry.models import NotificationSetting, User
+        from sentry.models import NotificationSetting
+        from sentry.services.hybrid_cloud.user import user_service
 
-        all_possible_users = User.objects.get_from_group(group)
-        active_and_disabled_subscriptions = self.filter(group=group, user__in=all_possible_users)
+        all_possible_users = user_service.get_from_group(group)
+        active_and_disabled_subscriptions = self.filter(
+            group=group, user_id__in=[u.id for u in all_possible_users]
```

Comment:
Hold on, you can actually avoid the service boundary altogether and avoid needing the `get_from_group` call.  That's because you only need the `user_id`, you don't need any other attribute of the user, and all the actually relevant models are in the region silo.

What about splitting `User.objects.get_from_group` up into two steps: one step , you find the org membership (which is local to the region) and return the relation (user_id, just the number).  The second step, fetches the users based on those user ids.

This greatly simplifies all this logic and prevents the unnecessary silo exchange (which become expensive in the end).

Reply:
Ah, that involves a lot of other code changes as well.
This seems brittle, hmm hmm hmm.
A LOT of time we use the full ORM object when all we mean is an id, and it's very very frustrating.

Reply:
Yeah, the "recipient" type switch matters in a _lot_ of places. Splitting up the notification system based on type would help a lot